elastic/kibana
# [Index Management] Fix allow_auto_create field in index template form

Fixes https://github.com/elastic/kibana/issues/177938

## Summary

This PR fixes the "Allow auto create" field in the index template form so that the default value is `undefined` instead of `false`.

Before:
<img width="1200" alt="Screenshot 2024-03-08 at 14 18 47" src="https://github.com/elastic/kibana/assets/59341489/c5edf280-f60d-49fe-9614-c8aa3a9e8737">


Now:
<img width="1209" alt="Screenshot 2024-03-08 at 14 06 43" src="https://github.com/elastic/kibana/assets/59341489/a887c4f6-52db-4609-904c-f8f60de8abca">

**How to test:**
1. Start Es and Kibana (these changes should be tested in both stateful and serverless).
2. Go to Stack Management -> Index Management -> Index Templates
3. Start creating a new index template.
4. Try changing the value in the "Allow auto create" field and verify that the value is correctly displayed in the last (Review) step, in the Summary tab.
5. After creating an index template, verify that the details flyout correctly displays the "Allow auto create" value.
6. Try editing a created index template and verify that the form contains the correct value in the "Allow auto create" field.


### Checklist

Delete any items that are not applicable to this PR.
